pkg/asset/manifests/openstack/cloudproviderconfig_test.go

```diff
@@ -49,46 +48,46 @@ func TestCloudProviderConfigSecretUserDomain(t *testing.T) {
 	}
 
 	expectedConfig := `[Global]
-auth-url    = https://my_auth_url.com/v3/
-username    = my_user
-password    = "my_secret_password"
-tenant-id   = f12f928576ae4d21bdb984da5dd1d3bf
-domain-id   = default
+auth-url = https://my_auth_url.com/v3/
+username = my_user
+password = "my_secret_password"
+tenant-id = f12f928576ae4d21bdb984da5dd1d3bf
+domain-id = default
 domain-name = Default
-region      = my_region
-
+region = my_region
 `
 	actualConfig, err := CloudProviderConfigSecret(&cloud)
 	assert.NoError(t, err, "failed to create cloud provider config")
 	assert.Equal(t, expectedConfig, string(actualConfig), "unexpected cloud provider config")
 }
 
 func TestCloudProviderConfigSecretQuoting(t *testing.T) {
-	cloud := clientconfig.Cloud{
-		AuthInfo: &clientconfig.AuthInfo{
-			Username:   "my_user",
-			Password:   "special \\r \\n \" \\ ",
-			AuthURL:    "https://my_auth_url.com/v3/",
-			ProjectID:  "f12f928576ae4d21bdb984da5dd1d3bf",
-			DomainID:   "default",
-			DomainName: "Default",
-		},
-		RegionName: "my_region",
+	passwords := map[string]string{
+		"regular":           "regular",
+		"with\\n":           "with\\\\n",
+		"with#":             "with#",
+		"with$":             "with$",
+		"with;":             "with;",
+		"with \r \n \" \\ ": "with \\r \\n \\\" \\\\ ",
+		"with!":             "with!",
+		"with?":             "with?",
+		"with`":             "with`",
 	}
 
-	expectedConfig := `[Global]
-auth-url    = https://my_auth_url.com/v3/
-username    = my_user
-password    = "special \\r \\n \" \\ "
-tenant-id   = f12f928576ae4d21bdb984da5dd1d3bf
-domain-id   = default
-domain-name = Default
-region      = my_region
+	for k, v := range passwords {
+		cloud := clientconfig.Cloud{
+			AuthInfo: &clientconfig.AuthInfo{
+				Password: k,
+			},
+		}
 
+		expectedConfig := `[Global]
+password = "` + v + `"
 `
-	actualConfig, err := CloudProviderConfigSecret(&cloud)
-	assert.NoError(t, err, "failed to create cloud provider config")
-	assert.Equal(t, expectedConfig, string(actualConfig), "unexpected cloud provider config")
+		actualConfig, err := CloudProviderConfigSecret(&cloud)
+		assert.NoError(t, err, "failed to create cloud provider config")
+		assert.Equal(t, expectedConfig, string(actualConfig), "unexpected cloud provider config")
+	}
```

Comment:
Since there is no standard for INI, we should make sure we're using the format our consumer expects. I would directly test that:

```Go
func TestCloudProviderConfigSecretQuoting(t *testing.T) {
	passwords := map[string]string{
		"regular":        "regular",
		"with\\n":        "with\\\\n",
		"with#":          "with#",
		"with$":          "with$",
		"with;":          "with;",
		"with \n \" \\ ": "with \\n \\\" \\\\ ",
		"with!":          "with!",
		"with?":          "with?",
		"with`":          "with`",
	}

	t.Run("expecting a literal", func(t *testing.T) {
		for k, v := range passwords {
			cloud := clientconfig.Cloud{
				AuthInfo: &clientconfig.AuthInfo{
					Password: k,
				},
			}

			expectedConfig := `[Global]
password = "` + v + `"
`
			actualConfig, err := CloudProviderConfigSecret(&cloud)
			assert.NoError(t, err, "failed to create cloud provider config")
			assert.Equal(t, expectedConfig, string(actualConfig), "unexpected cloud provider config")
		}
	})

	// We target gopkg.in/gcfg.v1
	t.Run("gcfg roundtrip", func(t *testing.T) {
		for want := range passwords {
			t.Run(want, func(t *testing.T) {
				// Encode to INI
				var b []byte
				{
					var err error
					b, err = CloudProviderConfigSecret(&clientconfig.Cloud{
						AuthInfo: &clientconfig.AuthInfo{
							Password: want,
						},
					})
					assert.NoError(t, err, "failed to encode")
				}

				// Decode to string
				var have string
				{
					var raw struct {
						Global struct{ Password string }
					}
					err := gcfg.ReadStringInto(&raw, string(b))
					assert.NoError(t, err, "failed to decode")
					have = raw.Global.Password
				}

				assert.Equal(t, want, have, "decoded value not matching the input")
			})
		}
	})
}
```

That's how I have discovered that `\r` is not an accepted escape sequence...